File: 2015/zig/1.zig
const std = @import("std");

pub fn main() !void {
    var args = std.process.args();
    _ = args.skip();
    var floors: i32 = 0;
    if (args.next()) |arg| {
        for (arg) |c| {
            switch (c) {
                '(' => floors += 1,
                ')' => floors -= 1,
                else => unreachable,
            }
        }
    }
    std.debug.print("floors: {}\n", .{floors});
}

File: 2015/rust/src/bin/1.rs
use std::os::unix::prelude::OsStrExt;

fn main() {
    let args = std::env::args_os();
    let input = unsafe { args.skip(1).next().unwrap_unchecked() };
    let mut floors = 0;
    for c in input.as_bytes() {
        match c {
            b'(' => floors += 1,
            b')' => floors -= 1,
            _ => panic!("Invalid input"),
        }
    }
    println!("Santa is on floor {}", floors);
}
